akka-testkit-typed/src/main/scala/akka/testkit/typed/TestKit.scala

```diff
@@ -64,12 +64,16 @@ class TestKit(name: String, config: Option[Config]) extends TestKitBase {
   def this(name: String) = this(name, None)
   def this(config: Config) = this(TestKit.getCallerName(classOf[TestKit]), Some(config))
   def this(name: String, config: Config) = this(name, Some(config))
+
   import TestKit._
-  implicit val system = ActorSystem(testKitGuardian, name, config = config)
+  implicit val system = ActorSystem(testKitGuardian, name, config = Some(config match {
+    case None                        ⇒ mixedInConfig.withFallback(ConfigFactory.load())
```

Comment:
I don't think the `withFallback` is needed here, the ActorSystem takes care of that

Reply:
AFAICS it is (https://github.com/akka/akka/blob/master/akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala#L176) - which is actually nice for testing, as it opens up the possibility of starting an ActorSystem that is not influenced by any `application.conf` on the classpath at all.

Reply:
yes, application.conf should not be loaded unless no config at all is specified when creating AS

Reply:
Right: so this approach now gives you 2 options when 'mixing in' configuration:

* If you want the effective config to be `mixedInConfig.withFallback(ConfigFactory.load())`, do not provide any configuration to `TestKit`
* If you want the effective config to be `explicitConfig.withFallback(mixedInConfig)`, provide the `explicitConfig` as the `TestKit` configuration

When you want the effective config to be something like `explicitConfig.withFallback(mixedInConfig).withFallback(ConfigFactory.load())`, you will have to provide it explicitly as `explicitConfig.withFallback(ManualTime.config).withFallback(ConfigFactory.load())`.

It does make the 'common case' (i.e. `ManualTimerSpec`) nice and simple, but the infrastructure is a bit complicated - hard to say if it is really worth it. I could move the 'mixedInConfig' logic out of this PR and to a separate one?

Reply:
> It does make the 'common case' (i.e. ManualTimerSpec) nice and simple, but the infrastructure is a bit complicated - hard to say if it is really worth it. I could move the 'mixedInConfig' logic out of this PR and to a separate one?

Maybe splitting off would be a good idea for now. The idea is good, though. It's worthwhile to introduce a principled way of stacking predefined settings (I guess one problem that will remain is that we cannot easily detect conflicts between different mixed-in config but that might be acceptable).